### PR TITLE
Issue 2846 ie8 failures

### LINKED ISSUE #2846: [ie8] Two test failures and an exception on IE8
This is with either login.dev.anosrep.org or transition.personatest.org. The exception happens well after the test failures.

```
failures: 
 [ 'dialog/js/modules/primary_offline: start controller with `add: false` and XHR error displays error screen (1, 0, 1)',
   'dialog/js/modules/set_password: create with transition_no_password (1, 1, 2)' ] }
```

Exception:

```
Can't move focus to the control because it is invisible, not enabled, or of a type that does not accept the focus. 
jquery-1.7.1.min.js, line 3 character 3117
```

### PR DESCRIPTION
**Pull request by shane-tomlinson**
@ozten - can you review this?

Fix IE8 errors
- Added a new test helpers - testElementTextContains.
- Use the new helper instead of text.indexOf to check for text in the DOM.
- Remove extra variables that were not used.
- Fix the thrown exception when focusing the password field in the unit tests.
- Remove the console.log message for `actions called={"doAuthenticate":true}`

fixes #2846
Partial fix for #2828

### CODE REVIEW

File: resources/static/test/cases/dialog/js/modules/set_password.js [hunk 1]
```diff
@@ -63,9 +64,9 @@
       email: "transition@password.no",
       transition_no_password: true
     });
-    var msg = $("#set_password .inputs li").text();
-    ok(msg.indexOf("no longer allows"), "transition message shown");
```
Comment (shane-tomlinson) [inline resources/static/test/cases/dialog/js/modules/set_password.js:67]: String.prototype.indexOf returns -1 if a value is not found. The index for this text in IE8 was 0.

File: resources/static/test/cases/dialog/js/misc/state.js [hunk 2]
```diff
@@ -480,7 +480,6 @@
       type: 'secondary',
       state: 'transition_to_secondary'
     });
-    console.log('actions called=' + JSON.stringify(actions.called));
```
Comment (ozten) [inline resources/static/test/cases/dialog/js/misc/state.js:483]: Doh. Thanks.

**Comment by ozten** [comment]
I'm r+, but you might want to do the following to the test suite:

1) In the testing framework, detect that we're using ie8 by simulating this failure and setting a `ie8` flag.

2) For all browsers, run through the steps, to catch dom exceptions, but

3) For tests which ie8 will get an empty string, skip checking for the expected string. Alternatively have a check for empty string, you're call.

**Comment by shane-tomlinson** [comment]
@ozten - Thanks for the review, I made two changes from your suggestions.

1) in testElementTextContains and testElementTextEquals, make sure there are no explosions if the elements to get the text from are not found.

2) Look more closely at the cause of the focus exception in IE8. The cause was due to a timing error where complete was called before the focus occurred. When complete called, the DOM got torn down before focus occurred. Since the element was no longer in the DOM, focus excepted. Since I moved complete into the showHint callback already, the problem was already side stepped and the try/catch was unnecessary. I removed it. This means no IE8 specific testing for the exception.

**Comment by ozten** [comment]
Thanks for jumping on 2852, great work!

**Pull request merged**